### PR TITLE
Display gender and age when examining humanoids

### PR DESCRIPTION
**Pull request by och-och**
## About the PR
I personally struggle with telling what other player's character's genders and preferred pronouns are. Even if I could tell what their sex is, it may not correspond with the pronouns they use (especially in case of they/it); and names are very often ambiguous. There have also been requests for such feature, so I'm not alone in this.

**Screenshots**
image

**Changelog**

:cl: och
- add: Examining people tells you their pronouns and approximate age

### CODE REVIEW

**Comment by Zumorica** [comment]
Honestly, just remove the "miserable pile of X" part, it's been there since at least 2018 and it's pretty unfunny.

File: Content.Shared/CharacterAppearance/Systems/SharedHumanoidAppearanceSystem.cs [hunk 1]
```diff
@@ -69,6 +74,28 @@ public void UpdateSexGender(EntityUid uid, Sex sex, Gender gender, HumanoidAppea
             RaiseLocalEvent(uid, new ChangedHumanoidAppearanceEvent(component.Appearance, component.Sex, component.Gender, component.Species), true);
         }
 
+        /// <summary>
+        /// Takes ID of the species prototype, returns UI-friendly name of the species.
+        /// </summary>
+        public string GetSpeciesRepresentation(string speciesId) {
+            var prototypeManager = IoCManager.Resolve<IPrototypeManager>();
```
Comment (mirrorcult) [inline Content.Shared/CharacterAppearance/Systems/SharedHumanoidAppearanceSystem.cs:81]: you dont need to manually resolve, store it as a `[Dependency]` in a system field

File: Content.Shared/CharacterAppearance/Systems/SharedHumanoidAppearanceSystem.cs [hunk 2]
```diff
@@ -69,6 +74,28 @@ public void UpdateSexGender(EntityUid uid, Sex sex, Gender gender, HumanoidAppea
             RaiseLocalEvent(uid, new ChangedHumanoidAppearanceEvent(component.Appearance, component.Sex, component.Gender, component.Species), true);
         }
 
+        /// <summary>
+        /// Takes ID of the species prototype, returns UI-friendly name of the species.
+        /// </summary>
+        public string GetSpeciesRepresentation(string speciesId) {
```
Comment (mirrorcult) [inline Content.Shared/CharacterAppearance/Systems/SharedHumanoidAppearanceSystem.cs:80]:
```suggestion
        public string GetSpeciesRepresentation(string speciesId) 
        {
```

and elsewhere. c# + codebase convention

File: Content.Shared/CharacterAppearance/Systems/SharedHumanoidAppearanceSystem.cs [hunk 3]
```diff
@@ -69,6 +74,28 @@ public void UpdateSexGender(EntityUid uid, Sex sex, Gender gender, HumanoidAppea
...
+            var prototypeManager = IoCManager.Resolve<IPrototypeManager>();
+            if (prototypeManager.TryIndex<SpeciesPrototype>(speciesId, out var species)) {
+                return Loc.GetString(species.Name);
+            }
+            else {
```
Comment (mirrorcult) [inline Content.Shared/CharacterAppearance/Systems/SharedHumanoidAppearanceSystem.cs:85]:
```suggestion
            if (prototypeManager.TryIndex<SpeciesPrototype>(speciesId, out var species)) 
            {
                return Loc.GetString(species.Name);
            }
            else 
            {
```

File: Content.Shared/CharacterAppearance/Systems/SharedHumanoidAppearanceSystem.cs [hunk 4]
```diff
@@ -69,6 +74,28 @@ public void UpdateSexGender(EntityUid uid, Sex sex, Gender gender, HumanoidAppea
...
+                return Loc.GetString("humanoid-appearance-component-unkown-species");
```
Comment (mirrorcult) [inline Content.Shared/CharacterAppearance/Systems/SharedHumanoidAppearanceSystem.cs:86]:
```suggestion
                return Loc.GetString("humanoid-appearance-component-unknown-species");
```

File: Resources/Locale/en-US/character-appearance/components/humanoid-appearance-component.ftl [hunk 5]
```diff
@@ -0,0 +1,2 @@
+humanoid-appearance-component-unkown-species = Person
```
Comment (mirrorcult) [inline Resources/Locale/en-US/character-appearance/components/humanoid-appearance-component.ftl:1]:
```suggestion
humanoid-appearance-component-unknown-species = Person
```

**Review by mirrorcult — CHANGES_REQUESTED** [review]

File: Content.Shared/CharacterAppearance/Systems/SharedHumanoidAppearanceSystem.cs (hunk 2 above)
Comment (och-och) [reply Content.Shared/CharacterAppearance/Systems/SharedHumanoidAppearanceSystem.cs]: Completely forgot this is how c# is written ☠️

**Review by mirrorcult — APPROVED** [review]

**Pull request merged**